python/vineyard/data/tests/test_dataframe.py

```diff
@@ -165,3 +211,56 @@ def test_dataframe_reusing(vineyard_client):
         meta1['__values_-value-0']['buffer_'].id
         == meta2['__values_-value-0']['buffer_'].id
     )
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        pd.DataFrame({'a': [1, 2, 3, 4], 'b': [5, 6, 7, 8]}),
+        pd.DataFrame({'a': ['1', '2', '3', '4'], 'b': ['5', '6', '7', '8']}),
+        pd.DataFrame([1, 2, 3, 4], columns=[['x']]),
+        pd.DataFrame({1: [1, 2, 3, 4], 2: [5, 6, 7, 8]}),
+        pd.DataFrame(
+            {
+                'a': [1, 2, 3, 4],
+                'b': [5, 6, 7, 8],
+                1: [9, 10, 11, 12],
+                2: [13, 14, 15, 16],
+            }
+        ),
+        pd.DataFrame(np.random.rand(10, 5), columns=['c1', 'c2', 'c3', 'c4', 'c5']),
+        pd.DataFrame(
+            [[1, 3, 3], [4, 2, 6], [7, 8, 9]],
+            index=['a1', 'a2', 'a3'],
+            columns=['x', 'y', 'z'],
+        ),
+        pd.arrays.SparseArray(np.random.randn(10)),
+        pd.DataFrame(np.random.randn(100, 4), columns=['x', 'y', 'z', 'a']).astype(
+            pd.SparseDtype("float", np.nan)
+        ),
+        pd.DataFrame(np.random.randn(100, 4), columns=[1, 2, 3, 4]).astype(
+            pd.SparseDtype("float", np.nan)
+        ),
+        pd.DataFrame(np.random.randn(100, 4), columns=['x', 'y', 'z', 0]).astype(
+            pd.SparseDtype("float", np.nan)
+        ),
+        pd.DataFrame(
+            pd.Series(
+                [
+                    pd.Timestamp("2012-05-01"),
+                    pd.Timestamp("2012-05-02"),
+                    pd.Timestamp("2012-05-03"),
+                ]
+            )
+        ),
+    ],
+)
+def test_with_ipc_and_rpc(value, vineyard_client, vineyard_rpc_client):
```

Comment:
Test what with ipc and rpc? Could you elaborate the function name to make it clear?

Reply:
If you want to run the test case with both IPC client and rpc client, use `pytest.mark.parametrize`.